### PR TITLE
Refactor checksum collection writing and usage

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

The first part of PR:

Instead of constructing a byte array with byte output stream and writers, and then later writing again
to a stream, we now get an OutputStream in and write directly to the stream

Additionally PR contains https://github.com/camunda/zeebe/pull/14184

- Support to update the checksum collection from bytes, which might be received or read from the file already. This is useful, if we want to avoid re-reading files etc.
- Support to persist snapshot with given checksum collection.
- collection of checksum is no longer recreated and directly written to the directory.

## Related issues

related #14045 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java [hunk 1]
```diff
@@ -61,11 +62,8 @@ private static boolean isNotMetadataFile(final Path file) {
 
   public static void persist(final Path checksumPath, final ImmutableChecksumsSFV checksum)
       throws IOException {
-    try (final RandomAccessFile checksumFile = new RandomAccessFile(checksumPath.toFile(), "rwd")) {
-      final byte[] data = checksum.serializeSfvFileData();
-      checksumFile.write(data);
-      checksumFile.setLength(data.length);
-    }
+    final FileOutputStream fileOutputStream = new FileOutputStream(checksumPath.toFile());
```
Comment (npepinpe) [inline snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java:65]: :x: Make sure to close the output stream

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java [hunk 2]
```diff
@@ -61,11 +62,8 @@ private static boolean isNotMetadataFile(final Path file) {
...
+    checksum.write(fileOutputStream);
```
Comment (npepinpe) [inline snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java:66]:
```suggestion
    try(final var fileOutputStream = new FileOutputStream(checksumPath.toFile())) {
      checksum.write(fileOutputStream);
    }
```

**Review by npepinpe — APPROVED** [review]
:+1: Just one comment, but no need for another review

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java (hunk 1 above)
Comment (Zelldon) [reply snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java:65]: I knew I was missing something 😆 thanks good catch

**Pull request merged**